### PR TITLE
Document delta saving feature.

### PR DESCRIPTION
**Pull request by mattstein**
I've attempted to describe the field and element type changes that can be made to implement delta saving. Hopefully without excessive verbosity.

### CODE REVIEW

File: extend/field-types.md [hunk 1]
```diff
@@ -32,3 +32,22 @@ class Plugin extends \craft\base\Plugin
     // ...
 }
 ```
+
+## Supporting Delta Saves
+
+If your field type needs to do anything following [`afterElementSave()`](api:craft\base\Field::afterElementSave()) or [`afterElementPropagate()`](api:craft\base\Field::afterElementPropagate()), you can improve performance by skipping processing when the field’s value is unchanged.
```
Comment (brandonkelly) [inline extend/field-types.md]:
```suggestion
If your field type does any processing from [afterElementSave()](api:craft\base\FieldInterface::afterElementSave()) or [afterElementPropagate()](api:craft\base\FieldInterface::afterElementPropagate()), you can improve performance by skipping processing when the field’s value is unchanged.
```

File: extend/element-types.md [hunk 2]
```diff
@@ -857,3 +857,24 @@ public function setEagerLoadedElements(string $handle, array $elements)
...
+Element forms can be configured to submit only the field values that actually changed on the page. This is a prerequisite to [field delta saves](/extend/field-types.md#supporting-delta-saves).
+
+If your element provides its own edit form template, here’s how you can configure it to submit delta field content:
+
+1. Enable delta input name registration at the top of your template.
+2. Add `registerDeltas: true` wherever you’ve used `_includes/fields.html` or `_includes/field.html`.
+
+```twig{1,6}
+{% do view.setIsDeltaRegistrationActive(true) %}
+
```
Comment (brandonkelly) [inline extend/element-types.md:872]: Maybe put a `...` line here surrounded by blank lines, so it’s more clear that these wouldn’t go right next to each other. (Don’t forget to change `{1,6}` to `{1,8}` in the process.)

File: extend/field-types.md [hunk 3]
```diff
@@ -32,3 +32,22 @@ class Plugin extends \craft\base\Plugin
...
+
+You can tell whether field content has changed by calling `isFieldDirty()` on the element.
```
Comment (brandonkelly) [inline extend/field-types.md]:
```suggestion
You can tell whether field content has changed by calling [isFieldDirty()](api:craft\base\ElementInterface::isFieldDirty()) on the element.
```

File: extend/field-types.md [hunk 4]
```diff
@@ -32,3 +32,22 @@ class Plugin extends \craft\base\Plugin
...
+
+This method checks for content changes before continuing to handle that element:
```
Comment (brandonkelly) [inline extend/field-types.md]: This line can probably be removed, as it’s not really adding anything.

**Review by brandonkelly — CHANGES_REQUESTED** [review]

**Review by mattstein — COMMENTED** [review]
All updated. Thanks @brandonkelly!

**Pull request closed without merging**